### PR TITLE
[MRG] Fix to size check.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4180

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def _save_split(epochs, fname, part_idx, n_parts):
 
     # write events out after getting data to ensure bad events are dropped
     data = epochs.get_data()
+    assert data.dtype == 'float64'
```
Comment (agramfort) [inline mne/epochs.py]: can you check that when creating an EpochsArray this is actually enforced?

Comment (jaeilepp) [inline mne/epochs.py]: Yeah, I checked. You want a test for it?

**Comment by agramfort** [comment]
good.

**Comment by agramfort** [comment]
thx @jaeilepp 

can you also backport?

**Comment by jaeilepp** [comment]
Done.

**Pull request merged**